Sources/SmokeDynamoDB/InMemoryDynamoDBCompositePrimaryKeyTableStore.swift

```diff
@@ -166,55 +166,64 @@ internal class InMemoryDynamoDBCompositePrimaryKeyTableStore {
         
         return EventLoopFuture.andAllSucceed(futures, on: eventLoop)
     }
-    
-    public func monomorphicBulkWriteWithoutThrowing<AttributesType, ItemType>(
+    public func monomorphicBulkWriteWithoutThrowing<AttributesType, ItemType>( 
         _ entries: [WriteEntry<AttributesType, ItemType>],
         eventLoop: EventLoop) -> EventLoopFuture<Set<BatchStatementErrorCodeEnum>> {
+            return monomorphicBulkWriteWithoutThrowingBatchStatementError(entries, eventLoop: eventLoop).map { errorSet in 
+                let codeErros: Set<BatchStatementErrorCodeEnum> = Set(errorSet.compactMap { $0.code })
+                return codeErros
+            }
+        }
+
+    public func monomorphicBulkWriteWithoutThrowingBatchStatementError<AttributesType, ItemType>(
+        _ entries: [WriteEntry<AttributesType, ItemType>],
+        eventLoop: EventLoop) -> EventLoopFuture<Set<BatchStatementError>> {
+            let batchStatementError = BatchStatementError(code: .duplicateitem, message: nil)
             
-            let futures = entries.map { entry -> EventLoopFuture<BatchStatementErrorCodeEnum?> in
+            let futures = entries.map { entry -> EventLoopFuture<BatchStatementError?> in
                 switch entry {
                 case .update(new: let new, existing: let existing):
                     return updateItem(newItem: new, existingItem: existing, eventLoop: eventLoop)
-                        .map { () -> BatchStatementErrorCodeEnum? in
+                        .map { () -> BatchStatementError? in
                             return nil
-                        }.flatMapError { error -> EventLoopFuture<BatchStatementErrorCodeEnum?> in
-                            let promise = eventLoop.makePromise(of: BatchStatementErrorCodeEnum?.self)
-                            promise.succeed(BatchStatementErrorCodeEnum.duplicateitem)
+                        }.flatMapError { error -> EventLoopFuture<BatchStatementError?> in
```

Comment:
As SwiftLint is saying, looks like you aren't using the error parameter and it can be replaced with _

Reply:
Fixed! Thanks